### DISCUSSION TITLE
New default libp2p config: light-client

### DISCUSSION THREAD

**Discussion by SgtPooki** [ipfs/helia#638] [Polls]
We currently have two configurations we use by default, assuming full libp2p node compatbility (barring browser/node divergences). I'm wondering if folks would find it useful to have a third libp2p config for browsers that don't need a full libp2p node, and only want http with direct retrieval?

related:
* https://github.com/aschmahmann/yellowstone-solana-explorer/pull/2#issuecomment-2366726135
* https://github.com/ipfs/ipld-explorer-components
* https://github.com/ipfs/ipfs-webui/issues/2266
* Helia's libp2p defaults:
    * [browser](https://github.com/ipfs/helia/blob/4f14996a9b976f2b60f4c8fe52a4fd1632420749/packages/helia/src/utils/libp2p-defaults.browser.ts#L37-L86)
    * [nodejs](https://github.com/ipfs/helia/blob/4f14996a9b976f2b60f4c8fe52a4fd1632420749/packages/helia/src/utils/libp2p-defaults.ts#L42-L96)

**Comment by 2color** [comment]
Can you link to the specific configurations?

**Reply by SgtPooki** [reply]
absolutely

**Comment by SgtPooki** [comment]
We might need to update [the current API](https://github.com/ipfs/helia/blob/4f14996a9b976f2b60f4c8fe52a4fd1632420749/packages/helia/src/index.ts#L98-L113) to support a callback so we can receive options like the current defaults do
